web-ui/src/components/member_selector/member_selector_dialog/MemberSelectorDialog.jsx

```diff
@@ -230,6 +254,22 @@ const MemberSelectorDialog = ({
         member => !selectedMembers.includes(member)
       );
 
+      // Exclude members that don't have the selected tenure.
+      if (tenure === Tenures.Custom) {
+        filteredMemberList = members.filter(member => {
+          const start = new Date(member.startDate);
+          return start <= customTenure;
+        });
+      } else if (tenure !== Tenures.All) {
```

Comment:
If a specific tenure value like "5 Years" was selected, keep only the members that have been at OCI at least that long.